### PR TITLE
[risk=low][RW-7089] Replace ProfileAccessModules with DataAccessPanel when DAR=true

### PR DESCRIPTION
**Pull request by jmthibault79**
Implementing https://projects.invisionapp.com/d/main#/console/21472057/455216387/preview

Clicking "Manage Data access" brings the user to the Data Access Requirements page, where equivalent functionality exists to complete modules.  This keeps the module completion code in one location.

If the CT is visible to users (e.g. in Test)

User has no access
no tiers

User has access to RT but not CT
RT-no-CT

User has access to both
yes-both

-----
If the CT is not visible to users (e.g. Prod)

User has RT access

yes-RT-only

User does not
no-RT

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/utils/access-tiers.tsx
Comment (jmthibault79) [inline ui/src/app/utils/access-tiers.tsx:25]: anything besides one of these 2 will result in `undefined`

File: ui/src/app/pages/profile/data-access-panel.spec.tsx
Comment (jmthibault79) [inline ui/src/app/pages/profile/data-access-panel.spec.tsx:13]: These conditions no longer apply to the simplified DataAccessPanel.

File: ui/src/app/pages/profile/data-access-panel.tsx
Comment (jmthibault79) [inline ui/src/app/pages/profile/data-access-panel.tsx:1]: This new version does a good deal less than the previous, which is an unlaunched implementation of a previous version of the mocks.  See https://precisionmedicineinitiative.atlassian.net/browse/RW-6256 and https://github.com/all-of-us/workbench/pull/4803 for the history.

We'll probably add equivalent functionality back in for https://precisionmedicineinitiative.atlassian.net/browse/RW-7059 once we've finalized the appearance/behavior.

File: ui/src/app/pages/profile/profile-component.tsx
Comment (jmthibault79) [inline ui/src/app/pages/profile/profile-component.tsx:118]: These are Promises.  Do we have a particular naming convention for these?

File: ui/src/app/pages/profile/profile-component.tsx
Comment (jmthibault79) [inline ui/src/app/pages/profile/profile-component.tsx:438]: There was a visual alignment bug which became more obvious with this change.  It's probably easier to view this file with whitespace diffs off.

File: ui/src/app/pages/profile/profile-component.tsx
Comment (jmthibault79) [inline ui/src/app/pages/profile/profile-component.tsx:445]: the original DataAccessPanel was never launched

**Comment by jmthibault79** [comment]
Waiting on PO+UX approval

**Comment by jmthibault79** [comment]
e2e failure is a notebooks test

File: ui/src/app/pages/profile/data-access-panel.tsx
Comment (als364) [reply ui/src/app/pages/profile/data-access-panel.tsx:1]: I do not have access to the new mocks but I believe you that this matches them 🤷

File: ui/src/app/pages/profile/data-access-panel.tsx
Comment (als364) [inline ui/src/app/pages/profile/data-access-panel.tsx:58]: Please use the react-router [Link](https://reactrouter.com/web/api/Link) for navigation instead.

Comment (jmthibault79) [reply ui/src/app/pages/profile/data-access-panel.tsx:58]: ok, thanks.

Do we always want this where `navigate` is currently used?

Comment (als364) [reply ui/src/app/pages/profile/data-access-panel.tsx:58]: Yes, we do. There's a low priority [ticket](https://precisionmedicineinitiative.atlassian.net/browse/RW-7168) to replace them all.

**Review by als364 — APPROVED** [review]

File: ui/src/app/pages/profile/data-access-panel.tsx
Comment (jmthibault79) [reply ui/src/app/pages/profile/data-access-panel.tsx]: updated here

**Pull request merged**
